### PR TITLE
feat: interactive commands using enquirer

### PR DESCRIPTION
**Pull request by akramsmahmoud**

### CODE REVIEW

**Comment by Cadienvan** [comment]
This should be rebased I guess?

**Comment by buggyzap** [comment]
Can we keep both args and interactive version? So we don't exclude any different implementation in advance.

If I use the command without args, going to interactive version.
If I use the args I can use it without prompts.

This way an user can execute the script in an automated way or use the interactive version.

Do you agree @akramsmahmoud @Cadienvan ?

**Comment by Cadienvan** [comment]
I love the idea! @akramsmahmoud what do you think?

File: index.mjs
Comment (Cadienvan) [inline index.mjs:14]: I would refactor it as follows (Or something similar):
```js
const params = [
  {
    type: 'input',
    name: 'repoUrl',
    message: 'Enter the url of the repo to analyze',
    required: true
  },
  {
    type: 'input',
    name: 'daysAmount',
    message: 'Enter the number of days to consider',
    required: true
  },
  {
    type: 'input',
    name: 'branchName',
    message: 'Enter the name of the branch to consider',
    initial: 'main',
    required: true
  },
  {
    type: 'select',
    name: 'outputFormat',
    message: 'Choose the output format',
    choices: ['json', 'html'],
    initial: 'json',
    required: true
  }
];

let useEnquirer = true;
if(process.argv.length === params.length + 2)
  useEnquirer = false;

let repoUrl, daysAmount, branchName, outputFormat;

if(useEnquirer) {
  const response = await prompt(params);
  repoUrl = response.repoUrl;
  daysAmount = response.daysAmount;
  branchName = response.branchName;
  outputFormat = response.outputFormat;
} else {
  const response = process.argv.slice(2);
  repoUrl = response[0] || params.find(p => p.name === 'repoUrl').initial || '';
  daysAmount = response[1] || params.find(p => p.name === 'daysAmount').initial || '';
  branchName = response[2] || params.find(p => p.name === 'branchName').initial || '';
  outputFormat = response[3] || params.find(p => p.name === 'outputFormat').initial || '';
}
```

So we can have a single configuration spreaded across both needs

File: README.md
Comment (Cadienvan) [inline README.md:15]: This would remain as before but I'd add a paragraph saying if you don't pass any parameter it will go in interactive mode.

**Review by Cadienvan — CHANGES_REQUESTED** [review]

**Comment by Cadienvan** [comment]
@buggyzap is been so kind to take this development under his arms implementing the requested changes in #8.
As this is, in fact, an issue he proposed to us I think it's correct for him to proceed, considering the fact this is standing still since 2 weeks ago, yet I still find meaningful and correct to include @akramsmahmoud in the reviewers list, so I'll close this one and wait for your approval there.

**Pull request closed without merging**
